File: 3447-clear-digits/clear-digits.py
class Solution:
    def clearDigits(self, s: str) -> str:
        s = list(s)

        while any(char.isdigit() for char in s):
            digit_index = next(i for i, char in enumerate(s) if char.isdigit())

            for j in range(digit_index - 1, -1, -1):
                if not s[j].isdigit():
                    s.pop(digit_index)
                    s.pop(j)
                    break
        return ''.join(s)

File: 3447-clear-digits/clear-digits.js
/**
 * @param {string} s
 * @return {string}
 */
var clearDigits = function(s) {
    const stack = [];
    for (let char of s) {
        if (/[a-z]/i.test(char)) {
            stack.push(char);
        } else if (/\d/.test(char)) {
            if (stack.length) stack.pop();
        }
    }
    return stack.join('');
};